tests/config/test_omegaconf_config.py

```diff
@@ -0,0 +1,381 @@
+    filepath.parent.mkdir(parents=True, exist_ok=True)
+    yaml_str = yaml.dump(config)
+    filepath.write_text(yaml_str)
+
+
+def _write_json(filepath: Path, config: Dict):
+    filepath.parent.mkdir(parents=True, exist_ok=True)
+    json_str = json.dumps(config)
+    filepath.write_text(json_str)
+
+
+def _write_dummy_ini(filepath: Path):
+    filepath.parent.mkdir(parents=True, exist_ok=True)
+    config = configparser.ConfigParser()
+    config["prod"] = {"url": "postgresql://user:pass@url_prod/db"}
+    config["staging"] = {"url": "postgresql://user:pass@url_staging/db"}
+    with filepath.open("wt") as configfile:  # save
+        config.write(configfile)
+
+
+@pytest.fixture
+def base_config(tmp_path):
+    filepath = str(tmp_path / "cars.csv")
+    return {
+        "trains": {"type": "MemoryDataSet"},
+        "cars": {
+            "type": "pandas.CSVDataSet",
+            "filepath": filepath,
+            "save_args": {"index": True},
+        },
+    }
+
+
+@pytest.fixture
+def local_config(tmp_path):
+    filepath = str(tmp_path / "cars.csv")
+    return {
+        "cars": {
+            "type": "pandas.CSVDataSet",
+            "filepath": filepath,
+            "save_args": {"index": False},
+        },
+        "boats": {"type": "MemoryDataSet"},
+    }
+
+
+@pytest.fixture
+def create_config_dir(tmp_path, base_config, local_config):
+    proj_catalog = tmp_path / _BASE_ENV / "catalog.yml"
+    local_catalog = tmp_path / _DEFAULT_RUN_ENV / "catalog.yml"
+    parameters = tmp_path / _BASE_ENV / "parameters.json"
+    project_parameters = dict(param1=1, param2=2)
+
+    _write_yaml(proj_catalog, base_config)
+    _write_yaml(local_catalog, local_config)
+    _write_json(parameters, project_parameters)
+
+
+@pytest.fixture
+def proj_catalog(tmp_path, base_config):
+    proj_catalog = tmp_path / _BASE_ENV / "catalog.yml"
+    _write_yaml(proj_catalog, base_config)
+
+
+@pytest.fixture
+def proj_catalog_nested(tmp_path):
+    path = tmp_path / _BASE_ENV / "catalog" / "dir" / "nested.yml"
+    _write_yaml(path, {"nested": {"type": "MemoryDataSet"}})
+
+
+use_config_dir = pytest.mark.usefixtures("create_config_dir")
+use_proj_catalog = pytest.mark.usefixtures("proj_catalog")
+
+
+class TestOmegaConfLoader:
+    @use_config_dir
+    def test_load_core_config_dict_get(self, tmp_path):
+        """Make sure core config can be fetched with a dict [] access."""
+        conf = OmegaConfLoader(str(tmp_path))
+        params = conf["parameters"]
+        catalog = conf["catalog"]
+
+        assert params["param1"] == 1
+        assert catalog["trains"]["type"] == "MemoryDataSet"
+        assert catalog["cars"]["type"] == "pandas.CSVDataSet"
+        assert catalog["boats"]["type"] == "MemoryDataSet"
+        assert not catalog["cars"]["save_args"]["index"]
+
+    @use_config_dir
+    def test_load_local_config(self, tmp_path):
+        """Make sure that configs from `local/` override the ones
+        from `base/`"""
```

Comment:
Could you explain how this is testing `local` override from `base`? It almost looks the same as the previous test.

Reply:
I've changed this so the first two tests focus on verifying the `.get()` and `[]` access and the third test actually verifies that `local` config overrides `base`.